### ISSUE TITLE
threads.js in Electron - Cannot find module '0.bundle.worker.js'

### ISSUE THREAD

**Issue by raz-sinay** [andywer/threads.js#194]
Hi,

I am trying to import a worker in electron (I am using electron-webpack) main thread and I get the following error:
Error: Cannot find module '0.bundle.worker.js'.

Does this library support electron main thread and electron renderer thread?

**Comment by andywer** [comment]
Hey Raz!

Threads.js works with electron, it uses node.js worker threads, though. The error above looks like something's wrong with the webpack setup. Do you use webpack with the `threads-plugin?

**Comment by raz-sinay** [comment]
I do, And I also think that the worker file is being processed with threads-plugin.
I'll try to make a basic example to share.

**Comment by andywer** [comment]
Why did you close the issue? 😉

**Comment by aviklai** [comment]
@andywer hi, I work with @raz-sinay and we are using electron-builder so we are using it's webpack configuration.
In order to fix the issue for our electron app I had to do 2 things:
1. In package.json - add the threads package to the whiteListedModules array.
2. Initialize the threads-plugin in as follows:
new ThreadsPlugin({ globalObject: `typeof self !== 'undefined' ? self : this` })

Maybe you can add this to the documentation for anyone else who works with electron-builder for their electron apps?

**Comment by andywer** [comment]
@aviklai Awesome! Would you like to test `threads-plugin@1.3.0-beta`? I changed the default for the worker `globalObject`, so it should work now without passing custom options to the plugin.

Will update the documentation to highlight the `whiteListedModules` field.

**Comment by aviklai** [comment]
@andywer thanks a lot! 
I am also trying to understand how we can use it in the renderer process as well and so far I was not able to make it work so if you have any idea, I will be glad to check it out and update.

**Comment by andywer** [comment]
@aviklai Yeah, there was a minor bug. Can you try `threads-plugin@1.3.0-beta.2`?

Renderer process… What didn't work / what error did you see? We are using threads.js in an electron renderer process over at https://github.com/satoshipay/solar, but we are using Parcel over there.

Still need to update the docs.

**Comment by aviklai** [comment]
@andywer The logs I see from the renderer process says "compiled with errors" and I also see the warning"No instantiations of threads.js workers found"

**Comment by andywer** [comment]
Sorry, I got sick this week. Will look into it soon.

Thanks already for investigating and opening the PR!

**Comment by andywer** [comment]
@aviklai I am back on my feet :) Can you please add this to `node_modules/threads/package.json`? It should fix selecting the wrong implementation:

```diff
"browser": {
+     "./dist/master/implementation.js": "./dist/master/implementation.browser.js",
+     "./dist/worker/implementation.js": "./dist/worker/implementation.browser.js",
+     "./dist-esm/master/implementation.js": "./dist-esm/master/implementation.browser.js",
+     "./dist-esm/worker/implementation.js": "./dist-esm/worker/implementation.browser.js",
```

> Regarding threads-plugin@1.3.0-beta.2 - it doesn't work. I get an error: SyntaxError: Unexpected identifier

That's weird. I didn't have time to test it in-depth yet, at least when I build it and just try to run the plugin script it doesn't cause any syntax errors… Gotta check in more detail later.

**Comment by raz-sinay** [comment]
@andywer,
(threads-plugin@1.3.0-beta.2 throws `SyntaxError: Unexpected identifier` for me too).

I've tested what you asked about and it works.
> "browser": {
>      "./dist/master/implementation.js": "./dist/master/implementation.browser.js",
>      "./dist/worker/implementation.js": "./dist/worker/implementation.browser.js",
>      "./dist-esm/master/implementation.js": "./dist-esm/master/implementation.browser.js",
>      "./dist-esm/worker/implementation.js": "./dist-esm/worker/implementation.browser.js",

However, I am still getting the warning:
> No instantiations of threads.js workers found.

We are bundling our app using [electron-webpack](https://webpack.electron.build). There seem to be some configuration setting which prevents the app from starting when it compiles with warnings. 

I've done some testing and I think it might be caused by the `afterCompile` event hook. I've console logged it everytime it is being dispatched and it seems to be dispatched multiple times.
The first dispatch happens before the threads-plugin have finished searching for worker instantiations, which causes the compilation process to end with warnings.

![image](https://user-images.githubusercontent.com/15093043/72827381-5bfe5b80-3c83-11ea-9c39-8ec1ddb75d3c.png)


I've replaced the `afterCompile` hook with the `[afterEmit](https://webpack.js.org/api/compiler-hooks/#afteremit)` hook and it seems to work.  I don't get the warning anymore, but I'm not sure it answers your requirements.
